chatter/chat-server/src/chat/chat.service.ts

```diff
@@ -7,155 +7,184 @@ import { Conversation } from './models/conversation.model';
+            { 'message.sendername': receiver, 'message.receivername': sender },
+          ],
+        },
+      },
+    ]);
 
-    /**
-     * marks receivers messages that are sent to the logged in user as read
-     * @param sender logged in user
-     * @param receiver receiver of logged in users messages
-     */
-    async markReceiverMessages(sender: string, receiver: string) {
-        const messages = await this.messageModel.aggregate([
-            { $unwind: '$message' },
+    if (messages.length > 0) {
+      try {
+        messages.forEach(async (messageContents: any) => {
+          await this.messageModel.updateOne(
             {
-                $match: {
-                    $and: [{ 'message.sendername': receiver, 'message.receivername': sender }],
-                },
+              'message._id': messageContents.message._id,
             },
-        ]);
-
-        if (messages.length > 0) {
-            try {
-                messages.forEach(async (messageContents: any) => {
-                    await this.messageModel.updateOne({
-                        'message._id': messageContents.message._id,
-                    }, {
-                        $set: {'message.$.isRead': true },
-                    });
-                });
-            } catch (err) {
-                throw new InternalServerErrorException('Error: marking receiver messages');
-            }
-        }
+            {
+              $set: { 'message.$.isRead': true },
+            },
+          );
+        });
+      } catch (err) {
+        throw new InternalServerErrorException(
+          'Error: marking receiver messages',
+        );
+      }
     }
+  }
 
-    /**
-     * check if chat conversation exists gets conversation or creates a new one
-     * creates and sends a message and updates both users chatList
-     * @param user logged in user
-     * @param senderId sender id
-     * @param receiverId receiver id
-     * @param receiverName receiver name
-     * @param message chat message
-     */
-    async sendMessage(user: User, senderId: string, receiverId: string, receiverName: string, message: string): Promise<Message> {
-        const conversations = await this.conversationModel.find({
-            $or: [
-                { senderId, receiverId },
-                { senderId: receiverId, receiverId: senderId },
-            ],
-        });
+  /**
+   * check if chat conversation exists gets conversation or creates a new one
+   * creates and sends a message and updates both users chatList
+   * @param user logged in user
+   * @param senderId sender id
+   * @param receiverId receiver id
+   * @param receiverName receiver name
+   * @param message chat message
+   */
+  async sendMessage(
+    user: User,
+    senderId: string,
+    receiverId: string,
+    receiverName: string,
+    message: string,
+  ): Promise<Message> {
+    if (message.length > this.MESSAGE_LENGTH)
+      message = message.substring(0, this.MESSAGE_LENGTH);
```

Comment:
throw a bad request instead of shortening the message

`if (message.length > this.MAX_MESSAGE_LENGTH)`
    `throw new BadRequestException({ message: `Error message occured ${error}`});`